src/shared/features/AppletSettings/ExportDataSetting/Popups/ExportSettingsPopup/ExportSettingsPopup.test.tsx

```diff
@@ -0,0 +1,130 @@
+import { page } from 'resources';
+import { DateFormats } from 'shared/consts';
+import { mockedAppletId, mockedApplet } from 'shared/mock';
+import { SettingParam, renderWithProviders } from 'shared/utils';
+
+import { ExportSettingsPopup } from './ExportSettingsPopup';
+import { DATA_TESTID_EXPORT_DATA_SETTINGS_POPUP } from '../../ExportDataSetting.const';
+import { ExportDateType } from '../../ExportDataSetting.types';
+
+const dateString = '2023-11-14T14:43:33.369902';
+const date = new Date(dateString);
+
+const preloadedState = {
+  applet: {
+    applet: {
+      ...initialStateData,
+      data: { result: { ...mockedApplet, createdAt: dateString } },
+    },
+  },
+};
+
+const mockOnClose = jest.fn();
+
+describe('ExportSettingsPopup', () => {
+  it('should appear if isOpen is true', async () => {
+    const mockOnExport = jest.fn();
+
+    renderWithProviders(
+      <ExportSettingsPopup isOpen onClose={mockOnClose} onExport={mockOnExport} />,
+      {
+        preloadedState,
+      },
+    );
+
+    await waitFor(() =>
+      expect(screen.getByTestId(DATA_TESTID_EXPORT_DATA_SETTINGS_POPUP)).toBeVisible(),
+    );
+  });
+
+  it('should not appear if isOpen is false', async () => {
+    const mockOnExport = jest.fn();
+
+    renderWithProviders(
+      <ExportSettingsPopup isOpen={false} onClose={mockOnClose} onExport={mockOnExport} />,
+      {
+        preloadedState,
+      },
+    );
+
+    expect(screen.queryByTestId(DATA_TESTID_EXPORT_DATA_SETTINGS_POPUP)).not.toBeInTheDocument();
+  });
+
+  it('should trigger onExport if download button is clicked', async () => {
+    const mockOnExport = jest.fn();
+
+    renderWithProviders(
+      <ExportSettingsPopup isOpen onClose={mockOnClose} onExport={mockOnExport} />,
+      {
+        preloadedState,
+      },
+    );
+
+    fireEvent.click(screen.getByText('Download CSV'));
+
+    expect(mockOnExport).toBeCalled();
+  });
+
+  describe('should appear export data popup for date range', () => {
+    test.each`
+      exportDataType              | description
+      ${ExportDateType.Last24h}   | ${'last 24h'}
+      ${ExportDateType.LastMonth} | ${'last month'}
+      ${ExportDateType.LastWeek}  | ${'last week'}
+      ${ExportDateType.AllTime}   | ${'all time'}
+    `('$description', async ({ exportDataType }) => {
+      const mockOnExport = jest.fn();
+
+      renderWithProviders(
+        <ExportSettingsPopup isOpen onClose={mockOnClose} onExport={mockOnExport} />,
+        {
+          preloadedState,
+        },
+      );
+      const dateType = screen.getByTestId(`${DATA_TESTID_EXPORT_DATA_SETTINGS_POPUP}-dateType`);
+      const input = dateType.querySelector('input');
+      input && fireEvent.change(input, { target: { value: exportDataType } });
+
+      fireEvent.click(screen.getByText('Download CSV'));
+      expect(mockOnExport).toBeCalled();
+    });
+  });
+
+  // prettier-ignore
+  describe('should appear export data popup for \'choose dates\' date range', () => {
```

Comment:
```suggestion
  describe("should appear export data popup for 'choose dates' date range", () => {
```
Please consider trying this approach and possibly removing //prettier-ignore. I believe it may be feasible after the recent changes to Prettier settings in the code.